connectors/src/connectors/google_drive/temporal/activities.ts

```diff
@@ -592,32 +595,52 @@ export async function renewWebhooks(pageSize: number) {
     },
     limit: pageSize,
   });
+
   for (const wh of webhooks) {
-    const connector = await Connector.findByPk(wh.connectorId);
-    if (connector) {
+    await renewOneWebhook(wh.id);
+  }
+
+  return webhooks.length;
+}
+
+export async function renewOneWebhook(webhookId: ModelId) {
+  const wh = await GoogleDriveWebhook.findByPk(webhookId);
+  if (!wh) {
+    throw new Error(`Webhook ${webhookId} not found`);
+  }
+
+  const connector = await Connector.findByPk(wh.connectorId);
+
+  if (connector) {
+    try {
       const webhookInfo = await registerWebhook(connector.connectionId);
       if (webhookInfo.isErr()) {
-        if (webhookInfo.error instanceof HTTPError) {
-          if (webhookInfo.error.statusCode === 401) {
-            // The user has revoked the access to the app.
-            // We don't have a way to handle that yet.
-            continue;
-          } else {
-            throw webhookInfo.error;
-          }
-        } else if (webhookInfo.error instanceof Error) {
-          throw webhookInfo.error;
-        }
+        throw webhookInfo.error;
       } else {
         await wh.update({
           webhookId: webhookInfo.value.id,
           expiresAt: new Date(webhookInfo.value.expirationTsMs),
         });
       }
+    } catch (e) {
+      logger.error(
+        { error: e },
+        `Failed to renew webhook for ${wh.connectorId}`
```

Comment:
logs should not be templated (to simplify search) wh.connectorId should be in the object above:
{ error: e, connectorId: wh.connectorId }

I would also add the workspaceId here

Reply:
My bad